### PR TITLE
Remove Media Explorer dependency

### PR DESCRIPTION
**Pull request by joshkadis**

### CODE REVIEW

**Comment by joshkadis** [comment]
@mviggiano Looking good so far, I need to complete this review but in the meantime can you look at the Travis results?

**Comment by mviggiano** [comment]
Travis issues fixed - lemme know what you think when you get the chance to finish up your review.

File: modules/class-simplechart-insert.php [hunk 1]
```diff
@@ -0,0 +1,302 @@
...
+		} else {
+			wp_send_json_success( false );
+		}
+	}
+
+	/**
+	 * Handles the AJAX request and returns an appropriate response. This should be used, for example, to perform an API request to the service provider and return the results.
+	 *
+	 * @param array $request The request parameters.
+	 * @return MEXP_Response|bool|WP_Error A MEXP_Response object should be returned on success, boolean false should be returned if there are no results to show, and a WP_Error should be returned if there is an error.
```
Comment (joshkadis) [inline modules/class-simplechart-insert.php]: `MEXP_Response`...

File: modules/class-simplechart-insert.php [hunk 2]
```diff
@@ -0,0 +1,302 @@
...
+	 */
+	// TO-DO: FIX THIS AND INCORPORATE AJAX_REQUEST
+	public function request( array $request ) {
+
+		// You'll want to handle connection errors to your service here. Look at the Twitter and YouTube implementations for how you could do this.
```
Comment (joshkadis) [inline modules/class-simplechart-insert.php]: ✂️

**Review by joshkadis — COMMENTED** [review]
Nice work overall! I'd prefer to break `simplechart-insert.js` and `class-simplechart-insert.php` into separate files for individual modules/classes.

More generally, I'd like to explore using the REST API instead of making a `wp.media.ajax` call handled by `'wp_ajax_simplechart_request'`. This pattern dates back to the bad old pre-4.7 days without the builtin REST API. If you think it's reasonably doable, let's go for it.

**Comment by mviggiano** [comment]
@joshkadis New commit has updated this PR pretty significantly. I've implemented a webpack build process in order to split out the js into individual modules and split up the PHP classes as well. I added ESLint to the webpack build and updated all of the relevant custom JS to this ticket to fit the extended Airbnb style guide standards. I left the existing js in `plugin.js` and `post-edit.js` alone for the moment - tackling those two re: ESLint standards is a job for another day, I think. I'm also not building scss->css with webpack yet; let me know if you want me to include that in this change as well or if we can make an additional ticket for that down the line.

I've also fixed all the travis issues specific to these code changes. The remaining issues are all fixed by your last commit in https://github.com/alleyinteractive/wordpress-simplechart/pull/89.

**Comment by mviggiano** [comment]
🐴

**Comment by joshkadis** [comment]
🐴  to you too!

**Pull request merged**